app/src/main/java/org/oppia/app/parser/StringToNumberWithUnitsParser.kt

```diff
@@ -0,0 +1,311 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.annotation.StringRes
+import org.oppia.app.R
+import org.oppia.app.model.Fraction
+import org.oppia.app.model.NumberUnit
+import org.oppia.app.model.NumberWithUnits
+import org.oppia.app.model.Unit
+import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Double
+import java.util.regex.Pattern
+
+/**
+ * This class contains methods that help to parse string to number, check realtime and submit time errors.
+ * reference https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.ts#L220.
```

Comment:
This reference no longer works (likely because the frontend scripts were recently moved). I suggest using something other than the develop branch (such as a branch hash) to create a permalink and avoid file moves breaking the reference later on.